src/mlpack/methods/matrix_completion/matrix_completion.hpp

```diff
@@ -0,0 +1,150 @@
+/**
+ * @file matrix_completion.hpp
+ * @author Stephen Tu
+ *
+ * A thin wrapper around nuclear norm minimization to solve
+ * low rank matrix completion problems.
+ */
+#ifndef __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_HPP
+#define __MLPACK_METHODS_MATRIX_COMPLETION_MATRIX_COMPLETION_HPP
+
+#include <mlpack/core/optimizers/lrsdp/lrsdp.hpp>
+
+namespace mlpack {
+namespace matrix_completion {
+
+/**
+ * This class implements the popular nuclear norm minimization heuristic for
+ * matrix completion problems. That is, given known values M_ij's, the
+ * following optimization problem (semi-definite program) is solved to fill in
+ * the remaining unknown values of X
+ *
+ *   min ||X||_* subj to X_ij = M_ij
+ *
+ * where ||X||_* denotes the nuclear norm (sum of singular values of X).
+ *
+ * For a theoretical treatment of the conditions necessary for exact recovery,
+ * see the following paper:
+ *
+ *   A Simpler Appoarch to Matrix Completion.
+ *   Benjamin Recht. JMLR 11.
+ *   http://arxiv.org/pdf/0910.0651v2.pdf
+ *
+ * An example of how to use this class is shown below:
+ *
+ * @code
+ * size_t m, n;        // size of unknown matrix
+ * arma::umat indices; // contains the known indices [2 x n_entries]
+ * arma::vec values;   // contains the known values [n_entries]
+ *
+ * MatrixCompletion mc(m, n, indices, values);
+ * mc.Recover();
+ * mc.Recovered();     // access completed matrix
+ * @endcode
+ *
+ * @see LRSDP
+ */
+class MatrixCompletion
+{
+ public:
+
+  /**
+   * Construct a matrix completion problem, specifying the maximum rank of the
+   * solution.
+   *
+   * @param m Number of rows of original matrix.
+   * @param n Number of columns of original matrix.
+   * @param indices Matrix containing the indices of the known entries (must be
+   *    [2 x p]).
+   * @param values Vector containing the values of the known entries (must be
+   *    length p).
+   * @param r Maximum rank of solution.
+   */
+  MatrixCompletion(const size_t m,
+                   const size_t n,
+                   const arma::umat& indices,
+                   const arma::vec& values,
+                   const size_t r);
```

Comment:
You could use `arma::sp_mat` instead of `arma::umat` and `arma::vec`.  I know that there are a few other classes that request sparse matrices in `umat`/`vec` format, so maybe it is best to provide both, if possible.